### ISSUE TITLE
Database backup fails command fails

### ISSUE THREAD

**Issue by sjelfull** [craftcms/cms#6328]
### Description

`Error backing up the database: The shell command "mysqldump --defaults-extra-file="/storage/runtime/temp/my.cnf" --add-drop-table --comments --create-options --dump-date --no-autocommit --routines --default-character-set=utf8 --set-charset --triggers --single-transaction --no-data --result-file="/storage/backups/site--2020-07-06-143825--v3.5.0-beta.3.sql" site && mysqldump --defaults-extra-file="/storage/runtime/temp/my.cnf" --add-drop-table --comments --create-options --dump-date --no-autocommit --routines --default-character-set=utf8 --set-charset --triggers --no-create-info --ignore-table=site.assetindexdata --ignore-table=site.assettransformindex --ignore-table=site.sessions --ignore-table=site.templatecaches --ignore-table=site.templatecachequeries --ignore-table=site.templatecacheelements --ignore-table=site.cache --ignore-table=site.templatecachecriteria site >> "/storage/backups/site--2020-07-06-143825--v3.5.0-beta.3.sql"" failed with exit code 5: mysqldump: [ERROR] mysqldump: Empty value for 'port' specified.`

### Additional info

- Craft version: 3.5.0-beta.3

**Comment by angrybrad** [comment]
@sjelfull Do you have a custom https://docs.craftcms.com/v3/config/config-settings.html#backupcommand set?

**Comment by sjelfull** [comment]
Nope, pretty vanilla config:

```php
<?
return [
    // Global settings
    '*' => [
        // Default Week Start Day (0 = Sunday, 1 = Monday...)
        'defaultWeekStartDay' => 1,

        // Whether generated URLs should omit "index.php"
        'omitScriptNameInUrls' => true,

        // Control Panel trigger word
        'cpTrigger' => 'admin',

        'convertFilenamesToAscii' => true,
        'addTrailingSlashesToUrls' => true,

        // The secure key Craft will use for hashing and encrypting data
        'securityKey' => getenv('SECURITY_KEY'),
        'devMode' => (bool)getenv('DEVMODE'),
        'enableGraphQlCaching' => (bool)getenv('ENABLE_GRAPHQL_CACHING'),

        // Whether to save the project config out to config/project.yaml
        // (see https://docs.craftcms.com/v3/project-config.html)
        'useProjectConfigFile' => true,

        'maxUploadFileSize' => '512M',

        'siteUrl' => getenv('SITE_URL'),

        'aliases' => [
            'siteRoot' => getenv('SITE_ROOT'),
            'siteUrl' => getenv('SITE_URL'),
            'siteUrlPublic' => getenv('SITE_URL_PUBLIC'),
            'imagesUrl' => getenv('ASSETS_URL') . 'images/',
            'imagesPath' => getenv('ASSETS_PATH') . 'images/',
            'documentsUrl' => getenv('DOCUMENTS_URL'),
            'documentsPath' => getenv('DOCUMENTS_PATH'),
        ],
    ],```

**Comment by angrybrad** [comment]
Are you specifying the port in your `.env`/`config/db.php` file at all?

**Comment by sjelfull** [comment]
I didn't, but I tried adding it after I hit this error. Made no difference.

**Comment by angrybrad** [comment]
Just tried to reproduce on the latest 3.5 release and can't, unfortunately. Are you using DB_DSN for your connection setting or manually specifying everything through the other DB_* settings?

Can you share what's in your `.env` and `config/db.php` redacting the sensitive bits?

**Comment by sjelfull** [comment]
This seems to have been resolved after upgrading to 3.5 RC1.

**Comment by angrybrad** [comment]
Ahh... if you're using DSN, the Craft is expecting the port to be explicitly supplied there.

    DB_DSN="mysql:host=localhost;dbname=db_name;port=3306"

Should fix it.

https://www.php.net/manual/en/ref.pdo-mysql.connection.php
